lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -293,16 +296,58 @@ instance Buildable (PartialTx era) where
         cardanoTxF :: Cardano.Tx era -> Builder
         cardanoTxF tx' = pretty $ pShow tx'
 
+-- validateUTxO :: UTxO -> TransactionLayer -> ...
+--
+data UTxOAssumptions = forall k ktype. UTxOAssumptions
+    { txLayer
+        :: TransactionLayer k ktype SealedTx
+        -- TODO: Replace with smaller and smaller parts of 'TransactionLayer'
+    , inputScriptLookup
+        :: Maybe ([(W.TxIn, W.TxOut)] -> [CA.Script KeyHash])
+    , inputScriptTemplate
+        :: Maybe ScriptTemplate
+    , description :: String
+    }
+
+instance Show UTxOAssumptions where
+    show (UTxOAssumptions _ _ template description) = unwords
+        [ "UTxOAssumptions {"
+        , "description = " <> show description
+        , "template = " <> show template
+        , "}"
+        ]
```

Comment:
What's the benefit of having a hand-written `Show` instance here, instead of just auto-deriving it?

If it's for testing purposes, perhaps we could define a `Buildable` instance instead?

Reply:
Yes, it's for counterexamples in the properties. It's not used in _this_ PR though, so I've dropped it for now.